src/pages/Todo2.jsx

```diff
@@ -0,0 +1,212 @@
+import { useState, useEffect } from "react";
+
+export default function Todo() {
+    const [searchText, setsearchText] = useState("");
+    const [tasks, settasks] = useState([
+      {
+        id: 1,
+        createdAt: new Date(),
+        updatedAt: new Date(),
+        status: "completed",
+        name: "Coding react",
+      },
+      {
+        id: 2,
+        createdAt: new Date(),
+        updatedAt: new Date(),
+        status: "pending",
+        name: "Prepare Nextjs",
+      },
+    ]);
+  
+    const [filteredResults, setfilteredResults] = useState([]);
+  
+    useEffect(() => {
+      if (searchText) {
+        const res = tasks.filter((e) =>
+          e.name.toLowerCase().includes(searchText.toLowerCase())
+        );
+        setfilteredResults(res);
+      } else {
+      }
+    }, [searchText]);
+  
+    const createTodo = (item) => {
+      settasks([...tasks, { id: tasks.length + 1, ...item }]);
+    };
+    const deleteTodo = (id) => {
+      settasks(tasks.filter((e) => e.id !== id));
+    };
+    const clear = (id) => {
+      settasks([]);
+    };
+  
+    const [todoToEdit, settodoToEdit] = useState(null);
+  
+    const editTodo = (todo) => {
+      const newTasks = tasks.map((task) =>
+        task.id === todo.id ? { ...task, ...todo } : task
```

Comment:
If I understand well, the `todo` is only the name and the id, so you don't need to repeat the id twice because in this case `task` has an id and you spread it up which is cool, and the `todo` comes also with an id and you are also spreading it up. What you should do if I'm not wrong is the following.
```suggestion
        task.id === todo.id ? { ...task, name: todo.name } : task
```
with the code above you keep the task props untouched and only update the name.

Reply:
Great  let me work on it.